frontend/model/contracts/mailbox.js

```diff
@@ -63,7 +63,7 @@ sbp('chelonia/defineContract', {
     },
     'gi.contracts/mailbox/setAttributes': {
       validate: (data, { state, meta }) => {
-        if (state.attributes.creator !== meta.username) {
+        if (state.attributes && state.attributes.creator !== meta.username) {
```

Comment:
I am realizing that if `state.attributes` is not defined, we are calling `'gi.actions/mailbox/setAttributes'`, but that assumes `state.attributes` is defined.

So something like this needs to be added to the `process`:

```js
      process ({ meta, data }, { state }) {
        // TODO: remove this hack before releasing version 1
        if (!state.attributes) {
          Vue.set(state, 'attributes', {})
        }
```

Reply:
That's a nice catch. Correct.